pkg/controller/nodemaintenance/drain.go

```diff
@@ -0,0 +1,190 @@
+package nodemaintenance
+
+import (
+	"fmt"
+	"math"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/kubernetes/pkg/kubectl/drain"
+)
+
+func runCordonOrUncordon(r *ReconcileNodeMaintenance, node *corev1.Node, desired bool) error {
+	cordonOrUncordon := "cordon"
+	if !desired {
+		cordonOrUncordon = "un" + cordonOrUncordon
+	}
+
+	log.Info(fmt.Sprintf("%s Node: %s", cordonOrUncordon, node.Name))
+
+	c := drain.NewCordonHelper(node)
+	if updateRequired := c.UpdateIfRequired(desired); updateRequired {
+		err, patchErr := c.PatchOrReplace(r.drainer.Client)
+		if patchErr != nil {
+			log.Error(err, fmt.Sprintf("Unable to %s Node %s \n", cordonOrUncordon, node.Name))
+		}
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func drainPods(r *ReconcileNodeMaintenance, nodeName string) error {
+	list, errs := r.drainer.GetPodsForDeletion(nodeName)
+
+	if errs != nil {
+		return utilerrors.NewAggregate(errs)
+	}
+
```

Comment:
You will have to start the informer exactly here. There is a highly theoritical chance otherwise that  the cache does not see a new pod which got added between the list call here and the informer start.